### ISSUE TITLE
Exclude test files in 'unexported-return' rule

### ISSUE THREAD

**Issue by semihbkgr** [mgechev/revive#1343]
The `unexported-return` rule doesn't make sense for functions in test files.
Test files are not part of the public API and are only accessible in other test files within the same package, regardless of whether types are public or not. We should consider excluding test files from this rule to avoid unnecessary warnings.

e.g:
```
main_test.go:6:15: exported func NewOpt returns unexported type *main.option, which can be annoying to use
```

It is also possible that the function can be changed to private without any effect, but I’m unsure whether the linter should warn about that.

**Comment by semihbkgr** [comment]
I've sent a PR. If you're okay with the proposed change, we can go ahead with it.

**Comment by ccoVeille** [comment]
> Thanks for the feedback!
> 
> A rule can be configured to [exclude test files](https://github.com/mgechev/revive#rule-level-file-excludes). To disable `unexported-return` checks in test files, you can add the following to your configuration:
> 
> ```toml
> [rule.unexported-return]
>    Exclude=["TEST"]
> ```
> 
> This should suppress the warning for cases like the one you mentioned in `main_test.go`.

I had no idea this was possible, is it supported somehow with golangci-lint config?

**Comment by alexandear** [comment]
> > Thanks for the feedback!
> > 
> > A rule can be configured to [exclude test files](https://github.com/mgechev/revive#rule-level-file-excludes). To disable `unexported-return` checks in test files, you can add the following to your configuration:
> > 
> > ```toml
> > [rule.unexported-return]
> >    Exclude=["TEST"]
> > ```
> > 
> > This should suppress the warning for cases like the one you mentioned in `main_test.go`.
> 
> I had no idea this was possible, is it supported somehow with golangci-lint config?

No, it's not. Golangci-lint has its own mechanism to exclude rules which is more powerful than revive's. See https://golangci-lint.run/usage/false-positives/
